pkg/build/build.go

```diff
@@ -0,0 +1,334 @@
+}
+
+func Write(outputPath string, result Result) error {
+	fmt.Println(line)
+	fmt.Println("building folder structure:")
+	fmt.Println(line)
+	err := os.MkdirAll(outputPath, 0744)
+	if err != nil {
+		return errors.New("could not create target folder")
+	}
+	i := 0
+	for _, folder := range result.Folders {
+		i++
+		folder = path.Join(outputPath, folder)
+		fmt.Println(i, folder)
+		err := os.MkdirAll(folder, 0744)
+		if err != nil {
+			return err
+		}
+	}
+	fmt.Println(line)
+	fmt.Println("writing files:")
+	fmt.Println(line)
+	i = 0
+	for file, res := range result.Files {
+		i++
+		file = path.Join(outputPath, file)
+		fmt.Println(res.Info.Mode().Perm(), i, file)
+		err := os.WriteFile(file, res.Data, res.Info.Mode().Perm())
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func readData(files []string) (interface{}, error) {
+	if len(files) == 0 {
+		return nil, nil
+	}
+	data := make(map[string]interface{})
+
+	for _, file := range files {
+		fileData := make(map[string]interface{})
+
+		dataBytes, err := os.ReadFile(file)
+		if err != nil {
+			return nil, errors.Wrapf(err, "could not read data file: %q", file)
+		}
+
+		var unmarshal func(data []byte, v interface{}) error
+		switch {
+		case strings.HasSuffix(file, ".json"):
+			unmarshal = json.Unmarshal
+		case strings.HasSuffix(file, ".yml") || strings.HasSuffix(file, ".yaml"):
+			unmarshal = yaml.Unmarshal
+		default:
+			return nil, errors.New("unsupported data file format i need .json, .yml or .yaml")
+		}
+
+		err = unmarshal(dataBytes, &fileData)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to unmarshal data from file %q", file)
+		}
+
+		for k, v := range fileData {
+			data[k] = v
+		}
+	}
+	return data, nil
+}
+
+func processFolder(folderPath string, data interface{}, templateFuncMap template.FuncMap) (result Result, err error) {
+	folderPath = path.Clean(folderPath)
+	ignore := getIgnore(folderPath)
+	if len(ignore) > 2 {
+		fmt.Println("found .bobignore, ignoring", strings.Join(ignore, ", "))
+	}
+	copiedFiles := getCopy(folderPath)
+	if len(copiedFiles) > 0 {
+		fmt.Println("found .bobcopy, copying", strings.Join(copiedFiles, ", "))
+	}
+	folders, err := getFolders(folderPath, ignore)
+	if err != nil {
+		return
+	}
+	p := Result{
+		Folders: folders,
+		Files:   map[string]FileResult{},
+	}
+	files, err := getFiles(folderPath, ignore)
+	if err != nil {
+		return Result{}, err
+	}
+	for _, file := range files {
+		run := true
+
+		for _, copyFile := range copiedFiles {
+			if strings.HasPrefix(file, copyFile) || file == copyFile {
+				run = false
+				break
+			}
+		}
+		p.Files[file], err = processFile(path.Join(folderPath, file), data, run, templateFuncMap)
+		if err != nil {
+			return p, err
+		}
+	}
+	return p, nil
+}
+
+func getIgnore(root string) (ignore []string) {
+	ignore = []string{".bobignore", ".bobcopy"}
+	return append(ignore, getStuff(root, ".bobignore")...)
+}
+
+func getStuff(root, name string) []string {
```

Comment:
what is stuff? please use correct name

Reply:
I'ts moved code, I promise i'll get back to refactor it at some point